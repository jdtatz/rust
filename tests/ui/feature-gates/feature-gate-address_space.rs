#[address_space(0)] //~ ERROR the `#[address_space]` attribute is an experimental feature
static FOO: usize = 42;

extern "C" {
    #[address_space(0)] //~ ERROR the `#[address_space]` attribute is an experimental feature
    static EXTERN_FOO: usize;
}

#[address_space(65_536)] //~ ERROR the `#[address_space]` attribute is an experimental feature
                         //~^ ERROR address space value in `address_space` is too large: `65536`
static TOO_BIG: usize = 7;

fn main() {}
