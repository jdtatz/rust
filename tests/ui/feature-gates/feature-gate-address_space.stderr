error[E0658]: the `#[address_space]` attribute is an experimental feature
  --> $DIR/feature-gate-address_space.rs:1:1
   |
LL | #[address_space(0)]
   | ^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #1 <https://github.com/rust-lang/rust/issues/1> for more information
   = help: add `#![feature(address_space)]` to the crate attributes to enable

error[E0658]: the `#[address_space]` attribute is an experimental feature
  --> $DIR/feature-gate-address_space.rs:5:5
   |
LL |     #[address_space(0)]
   |     ^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #1 <https://github.com/rust-lang/rust/issues/1> for more information
   = help: add `#![feature(address_space)]` to the crate attributes to enable

error[E0658]: the `#[address_space]` attribute is an experimental feature
  --> $DIR/feature-gate-address_space.rs:9:1
   |
LL | #[address_space(65_536)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #1 <https://github.com/rust-lang/rust/issues/1> for more information
   = help: add `#![feature(address_space)]` to the crate attributes to enable

error: address space value in `address_space` is too large: `65536`
  --> $DIR/feature-gate-address_space.rs:9:1
   |
LL | #[address_space(65_536)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the value may not exceed `u16::MAX`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0658`.
